### ISSUE TITLE
About dice loss

### ISSUE THREAD

**Issue by sagarax009** [jakeret/tf_unet#20]
Hi 
I am just new to this area.I am using dice loss function for image segmentation. I found different presentation so got confused. Can you please explain me why there 1- 

for example : if my network for batch size 1 gives me values Top =0.9 ,bottom =0.9 then reduce mean is 0.9 and loss is 0.1 as per below equation. 

loss = 1 - tf.reduce_mean(2 * intersection/ (union))

But I found some codes ,they are just using 
loss =np.sum(2*intersection/(union)) 
so can you please clear this confusion?

**Comment by jakeret** [comment]
Not sure if my thought were correct back then, so any feedback is welcome.
The dice coefficient reaches 1 as best value (full aggreement). Hence, we want to reduce the difference between the current score and the best achiveable value:
`lossFunction = 1 - currentScore`

**Comment by sagarax009** [comment]
I agree with your point, I am convinced till dice coefficient calculation , but when batch contains two volumes one is label and one is image then what I should do ? Take mean or sum then subtract from 1.
currently I am getting following plot so i am little bit confused
![loss_crossing1](https://cloud.githubusercontent.com/assets/20017611/23586747/33462a12-019c-11e7-8c0a-78d5fb02f51a.png)

**Comment by jakeret** [comment]
This looks indeed a bit odd. (btw, you could use the Tensorboard so that you don't have to plot the metrics by yourself).

> when batch contains two volumes one is label and one is image

not quite sure what you mean. Could you clarify a bit? What do you mean exaclty with batch and volume?

**Comment by sagarax009** [comment]
Ohh sorry for confusion I am using caffe and in caffe you pass data into batch ,volume is just what is size of your layer for ex (124*64*64) . But I am using dice loss layer, the code I am referring they are using 


loss =np.sum(2*intersection/(union))

where as in your case its 

loss = 1 - tf.reduce_mean(2 * intersection/ (union))

here is conflict, because the loss function is same but different approach to get loss value but with sum I am not getting desired results. Hope you got me now

**Comment by jakeret** [comment]
I just pushed a new [branch](https://github.com/jakeret/tf_unet/tree/dice_coeff) where I changed the computation of the dice coefficient.
I'm also using the predictions instead of the logits as suggested in #15 . I tested it with the toy example of `tf_unet`.

**Comment by jakeret** [comment]
I merged the branch into master. Hope this resolves the issue
